packages/clay-date-picker/src/DayNumber.tsx

```diff
@@ -7,43 +7,80 @@ import {Keys} from '@clayui/shared';
 import classnames from 'classnames';
 import React from 'react';
 
-import {IDay, formatDate, setDate} from './Helpers';
+import {IDay, setDate} from './Helpers';
 
 interface IProps {
 	day: IDay;
-	daySelected: Date;
+	daysSelected: readonly [Date, Date];
 	disabled?: boolean;
+	range?: boolean;
 	onClick: (date: Date) => void;
 }
 
+interface IInterval {
+	start: Date;
+	end: Date;
+}
+
+function isWithinInterval(dirtyDate: Date, interval: IInterval) {
+	const time = dirtyDate.getTime();
+	const startTime = interval.start.getTime();
+	const endTime = interval.end.getTime();
+
+	if (startTime > endTime) {
+		return false;
+	}
+
+	return time >= startTime && time <= endTime;
+}
+
 const ClayDatePickerDayNumber: React.FunctionComponent<IProps> = ({
 	day,
-	daySelected,
+	daysSelected,
 	disabled,
 	onClick,
+	range,
 }) => {
 	const {date, outside} = day;
+	const [startDate, endDate] = daysSelected;
+
+	const hasEndDateSelected = date.toDateString() === endDate.toDateString();
+	const hasStartDateSelected = date.toDateString() === startDate.toDateString();
 
 	const classNames = classnames(
 		'date-picker-date date-picker-calendar-item',
 		{
-			active: date.toDateString() === daySelected.toDateString(),
+			active: hasStartDateSelected || (range && hasEndDateSelected),
 			disabled: outside || disabled,
 		}
 	);
 
 	return (
-		<div className="date-picker-col">
+		<div
+			className={classnames(
+				'date-picker-col',
+				range && {
+					'c-selected':
+						!hasStartDateSelected &&
+						!hasEndDateSelected &&
+						isWithinInterval(date, {
+							end: endDate,
+							start: startDate,
+						}),
+					'c-selected c-selected-end':
+						hasEndDateSelected && !hasStartDateSelected,
+					'c-selected c-selected-start':
+						hasStartDateSelected && !hasEndDateSelected,
+				}
+			)}
+		>
 			<button
-				aria-label={formatDate(
-					setDate(date, {
-						hours: 12,
-						milliseconds: 0,
-						minutes: 0,
-						seconds: 0,
-					}),
-					'yyyy MM dd'
-				)}
+				aria-label={setDate(date, {
+					hours: 12,
+					milliseconds: 0,
+					minutes: 0,
+					seconds: 0,
+				}).toDateString()}
```

Comment:
The only problem with using it like this is that we will have problems with internationalization.

Reply:
I forgot to test when changing browser's language. I'll take a look.

Reply:
Yes, according [a website](https://www.globalyzer.com/gzserver/help/reference/localeSensitiveMethods/js_toDateString.html) when using `toDateString` We may enter in some i18n issues.[ I found an utility for it on date-fns](https://date-fns.org/v2.20.3/docs/I18n), we can pass a locale to the format function. [However, We need to provide the locale in `ISO 639-1`](https://date-fns.org/v2.20.3/docs/Locale).